cartographer_ros/cartographer_ros/assets_writer_main.cc

```diff
@@ -205,25 +168,51 @@ void Run(const string& pose_graph_filename,
       const double duration_in_seconds =
           (view.getEndTime() - begin_time).toSec();
 
+      // We make sure that tf_messages are published before any data messages,
+      // so that tf lookups always work and that tf_buffer has a small cache
+      // size - because it gets very inefficient with a large one.
+      std::deque<rosbag::MessageInstance> delayed_messages;
```

Comment:
Please explain the motivation for introducing the delayed messages.

Reply:
This is following the offline node. Also see #349.